src/server/bitops_family.cc

```diff
@@ -0,0 +1,267 @@
+  (*cntx)->SendLong(0);
+}
+
+void BitCount(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitField(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitFieldRo(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitOp(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendOk();
+}
+
+void GetBit(CmdArgList args, ConnectionContext* cntx) {
+  // Support for the command "GETBIT key offset"
+  // see https://redis.io/commands/getbit/
+
+  int32_t offset{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto cb = [&](Transaction* t, EngineShard* shard) { return ReadValue(t->GetOpArgs(shard), key); };
+  Transaction* trans = cntx->transaction;
+  OpResult<std::string> result = trans->ScheduleSingleHopT(std::move(cb));
+
+  if (result) {
+    DVLOG(2) << "GET" << trans->DebugId() << "': key: '" << key << ", value '" << result.value()
+             << "'\n";
+    // we have the value, now we need to get the bit at the location
+    long val = GetBitValueSafe(result.value(), offset);
+    (*cntx)->SendLong(val);
+  } else {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      default:
+        DVLOG(2) << "GET " << key << " nil";
+        (*cntx)->SendLong(0);  // in case we don't have the value we should just send 0
+    }
+  }
+}
+
+void SetBit(CmdArgList args, ConnectionContext* cntx) {
+  int32_t offset{0};
+  int32_t value{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset) || !absl::SimpleAtoi(ArgS(args, 3), &value)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto index = cntx->db_index();
+  auto memcache_flags = cntx->conn_state.memcache_flag;
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    OverRideValue set_operation{t->GetOpArgs(shard), index, memcache_flags};
+
+    return set_operation.Set(key, offset, value);
+  };
+
+  Transaction* trans = cntx->transaction;
+  OpResult<long> result = trans->ScheduleSingleHopT(std::move(cb));
+  if (result) {
+    (*cntx)->SendLong(result.value());
+  } else {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      case OpStatus::OUT_OF_MEMORY:
+        (*cntx)->SendError(kOutOfMemory);
+        break;
+      default:
+        DVLOG(2) << "SETBIT " << key << " nil" << result.status();
+        (*cntx)->SendLong(0);  // in case we don't have the value we should just send 0
+        break;
+    }
+  }
+}
+
+// ------------------------------------------------------------------------- //
+// This are the "callbacks" that we're using from above
+std::string GetString(EngineShard* shard, const PrimeValue& pv) {
+  std::string res;
+  if (pv.IsExternal()) {
```

Comment:
I think it's ok not to copy tiered storage code, just add `CHECK(!pv.IsExternal())`.